dot-net-sdk/dto/EppoValue.cs

```diff
@@ -1,60 +1,93 @@
 using System.Globalization;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NLog;
 
 namespace eppo_sdk.dto;
 
 [JsonConverter(typeof(EppoValueDeserializer))]
 public class EppoValue
 {
-    public string? value { get; set; }
+    public string? Value { get; set; }
 
-    public EppoValueType type { get; set; } = EppoValueType.NULL;
+    public EppoValueType Type { get; set; } = EppoValueType.NULL;
 
-    public List<string>? array { get; set; }
+    private List<string>? _array;
 
+    private JObject? _json;
 
     public static EppoValue Bool(string? value) => new(value, EppoValueType.BOOLEAN);
     public static EppoValue Bool(bool value) => new(value.ToString(), EppoValueType.BOOLEAN);
     public static EppoValue Number(string value) => new(value, EppoValueType.NUMBER);
     public static EppoValue String(string? value) => new(value, EppoValueType.STRING);
     public static EppoValue Integer(string value) => new(value, EppoValueType.INTEGER);
+    public static EppoValue JsonObject(string value) => new(value, EppoValueType.JSON);
     public static EppoValue Null() => new();
 
+
+    private static readonly Logger Logger = LogManager.GetLogger("EppoValue");
+
     public EppoValue()
     {
     }
 
     public EppoValue(string? value, EppoValueType type)
     {
-        this.value = value;
-        this.type = type;
+        this.Value = value;
+        this.Type = type;
+
+        // This is helpful but parsing will actually not recognize JSON; it will merely see a string.
+        if (type == EppoValueType.JSON)
+        {
+            _json = JObject.Parse(value);
+        }
```

Comment:
@typotter I'm sorry for not clarifying this in the requirements - there is a encoded JSON in the API response type as well as the underlying JSON. Here is an example:

```
➜  curl --location '/xyz/api/randomized_assignment/v3/config?apiKey=xyz'
{
    "flags": {
        "json-flag": {
            "enabled": true,
            "allocations": {
                "allocation-33": {
                    "percentExposure": 1,
                    "variations": [
                        {
                            "name": "json1",
                            "variationKey": "variation-153",
                            "value": "{\n    \"foo\": \"bar\"\n}",
                            "typedValue": {
                                "foo": "bar"
                            },
                            "algorithmType": "CONSTANT",
                            "shardRange": {
                                "start": 0,
                                "end": 10000
                            }
                        }
                    ],
                    "statusQuoVariationKey": null,
                    "shippedVariationKey": null,
                    "holdouts": []
                }
            },
            "rules": [
                {
                    "allocationKey": "allocation-33",
                    "conditions": []
                }
            ],
            "subjectShards": 10000,
            "overrides": {},
            "typedOverrides": {}
        }
    }
}
```

I think it is fine to keep it as-is and migrate it to the `typedValue` as part of the UFC work - in that newer API the stringified value won't be returned.

Reply:
This makes things... interesting. Sometimes an EppoValue is encoded and comes with a `typedValue` which is not (as in the case of `Variation`). Other times (i.e. `Condition`), the `value` field is already typed.
The good news is that I don't, in fact, need to double deserialize anything. The bad news is that I just ended up starting all over with a different approach - one that will be much easier to transition to UFC and will also be nearly a drop-in swap to migrate to from `Newtonsoft.JSON` to the preferred `System.Text.JSON`